File: secure-skeleton/src/main/java/br/avcaliani/secureskeleton/user/controllers/UserController.java
package br.avcaliani.secureskeleton.user.controllers;

import br.avcaliani.secureskeleton.user.model.entities.User;
import br.avcaliani.secureskeleton.user.services.IUserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController("/user")
public class UserController {

    @Autowired
    private IUserService userService;

    @PostMapping("/sign-up")
    public User signUp(@RequestBody User user) throws Exception {
        return this.userService.save(user);
    }

}

File: kotlin-app/src/main/kotlin/br/avcaliani/kotlin/app/controller/UserController.kt
package br.avcaliani.kotlin.app.controller

import br.avcaliani.kotlin.app.model.User
import br.avcaliani.kotlin.app.service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/user")
class UserController {

    @Autowired
    private lateinit var service: UserService

    @GetMapping
    fun find(): List<User> = service.find()

    @GetMapping("/{id}")
    fun find(@PathVariable id: String): User = service.find(id)

    @PostMapping
    fun add(@RequestBody user: User): User = service.save(user)

    @DeleteMapping("/{id}")
    fun remove(@PathVariable id: String): Unit = service.remove(id)

}